pkg/operator/vmware.go

```diff
@@ -38,6 +38,13 @@ func (c *vSphereProblemDetectorController) connect(ctx context.Context) (*vspher
 
 	vmClient, err := c.newClient(ctx, cfg, username, password)
 	if err != nil {
+		if strings.Index(username, "\n") != -1 {
+			return nil, nil, fmt.Errorf("failed to connect to %s: username in credentials contains new line", cfg.Workspace.VCenterIP)
+		}
+
+		if strings.Index(password, "\n") != -1 {
+			return nil, nil, fmt.Errorf("failed to connect to %s: password in credentials contains new line", cfg.Workspace.VCenterIP)
```

Comment:
Did we really just want to log it? I thought we want to emit a metric via newly introduced sync_error - https://github.com/openshift/vsphere-problem-detector/pull/39/files#diff-a3f56fc9b14fd82bffcb6718ae74b90cb66ae96297c8c010e9fba4e2d19a6344R53

Reply:
For sure, I'll add that.